packages/flutter_tools/test/src/ios/devices_test.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' show ProcessResult;
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/ios/devices.dart';
+
+import 'package:mockito/mockito.dart';
+import 'package:process/process.dart';
+import 'package:test/test.dart';
+
+import '../context.dart';
+class MockProcessManager extends Mock implements ProcessManager {}
+class MockFile extends Mock implements File {}
+
+void main() {
+  group('test screenshot', () {
+    MockProcessManager mockProcessManager;
+    MockFile mockOutputFile;
+    IOSDevice iosDeviceUnderTest;
+
+    setUp(() {
+      mockProcessManager = new MockProcessManager();
+      mockOutputFile = new MockFile();
+    });
+
+    testUsingContext(
+      'screenshot without ideviceinstaller error',
+      () async {
+        when(mockOutputFile.path).thenReturn('/some/test/path/image.png');
```

Comment:
nit: here and below: even though I suspect we'll never support iOS screenshotting on Windows, and this is a mock either way probably a good idea to to use `path.join()` to keep the codebase as cross-platform as possible.